dist/tools/compile_test/compile_test.py

```diff
@@ -100,17 +100,19 @@ def build_all():
             stdout.write('\tBuilding application: {} ({}/{}) '.format(colorize_str(application, Termcolor.blue), nth, len(applications)))
             stdout.flush()
             try:
+                app_dir = join(riotbase, folder, application)
                 subprocess = Popen(('make', 'buildtest'),
                                    bufsize=1, stdin=null, stdout=PIPE, stderr=null,
-                                   cwd=join(riotbase, folder, application),
+                                   cwd=app_dir,
                                    env=subprocess_env)
 
                 results, results_with_output = tee(get_results_and_output_from(subprocess.stdout))
                 results = groupby(sorted(results), lambda res: res[0])
                 results_with_output = list(filter(lambda res: res[2].getvalue(), results_with_output))
                 failed_with_output = list(filter(lambda res: 'failed' in res[0], results_with_output))
                 success_with_output = list(filter(lambda res: 'success' in res[0], results_with_output))
-                check_call(["rm", "-rf", join(riotbase, folder, application, "bin")])
+                if join(app_dir, "bin") not in environ.get("PATH", "/bin:/usr/bin:/usr/local/bin:"):
```

Comment:
Smart solution :+1: 
Could you add a one line comment what this test does for someone who might not know why it is a bad idea to blindly `rm -rf` stuff named `bin`

Reply:
Done.